guiEditor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -198,13 +188,36 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
             description: "",
             tags: "",
         };
-
         xmlHttp.send(JSON.stringify(dataToSend));
     }
 
+    saveToSnippetServer = async () => {
+        const adt = this.props.globalState.guiTexture;
+        const content = JSON.stringify(adt.serializeContent());
+        let success = null;
+        if (this.props.globalState.customSave) {
+            await this.props.globalState.customSave!.action(content).then((resolve) => {
+                adt.snippetId = (resolve as unknown as string);
+                success = true;
+            }).catch((err) => {
+                alert(err);
+            });
+        }
+        else {
+            while (success === null) {

```

Comment:
This will block execution. why isn't `saveToSnippetServerHelper` return promise as well? 
then you do - 

```
const savePromise = this.props.globalState.customSave?.action || this.saveToSnippetServerHelper;
savePromise(content).then((resolve) => {
                adt.snippetId = (resolve as unknown as string);
                success = true;
            }).catch((err) => {
                alert(err);
            });
```
let's not use a while loop to control async events :-)